packages/graph/hash_graph/lib/graph/src/store/postgres/knowledge/entity/mod.rs

```diff
@@ -372,45 +385,82 @@ impl<C: AsClient> PostgresStore<C> {
 impl<C: AsClient> EntityStore for PostgresStore<C> {
     async fn create_entity(
         &mut self,
-        properties: EntityProperties,
-        entity_type_id: VersionedUri,
-        owned_by_id: OwnedById,
-        entity_uuid: Option<EntityUuid>,
+        entity_id: EntityId,
+        decision_time: Option<Timestamp>,
         updated_by_id: UpdatedById,
+        archived: bool,
+        entity_type_id: VersionedUri,
+        properties: EntityProperties,
         link_data: Option<LinkData>,
     ) -> Result<EntityMetadata, InsertionError> {
-        let transaction = PostgresStore::new(
-            self.as_mut_client()
-                .transaction()
-                .await
-                .into_report()
-                .change_context(InsertionError)?,
-        );
+        let entity_type_version_id = self
+            .version_id_by_uri(&entity_type_id)
+            .await
+            .change_context(InsertionError)?;
 
-        let entity_uuid = entity_uuid.unwrap_or_else(|| EntityUuid::new(Uuid::new_v4()));
-        let entity_id = EntityId::new(owned_by_id, entity_uuid);
+        let properties = serde_json::to_value(properties)
+            .into_report()
+            .change_context(InsertionError)?;
 
-        // TODO: match on and return the relevant error
-        //   https://app.asana.com/0/1200211978612931/1202574350052904/f
-        transaction.insert_entity_uuid(entity_uuid).await?;
-        let metadata = transaction
-            .insert_entity(
-                entity_id,
-                properties,
-                entity_type_id,
-                updated_by_id,
-                link_data,
+        let row = self
+            .as_client()
+            .query_one(
+                r#"
+                SELECT
+                    entity_edition_id,
+                    lower(decision_time),
+                    NULLIF(upper(decision_time), 'infinity'),
+                    lower(system_time),
+                    NULLIF(upper(system_time), 'infinity')
+                FROM
+                    create_entity($1, $2, $3, $4, $5, $6, $7, $8, $9, $10, $11, $12, $13);
```

Comment:
An unfortunate side-effect of shoving this in a stored-proc is it makes it _**very**_ hard for us to sanity check the ordering of parameters. Can you think of anything we can do here to make it better? @thehabbos007 any thoughts?

Reply:
Postgres supports named parameter calling notation for functions, but not sure if this is restricted to the built in ones only. It would look something like:
```sql
create_entity(_owned_by_id => $1)
```
https://www.postgresql.org/docs/current/sql-syntax-calling-funcs.html#SQL-SYNTAX-CALLING-FUNCS-NAMED

Reply:
If there aren't meaningful restrictions with doing so, I'd be more comfortable with that. Shoving 13 unnamed things in a function is not fun

Reply:
Changed in 9064133f29e9b3f8349d1d9ab80a9d327827d2bc

Reply:
Damn that does add a lot of lines. I think it's a lot better though, thank you!